src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(health = 100, name) {
+    this.health = health;
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(health, name) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(health, name) {
+    super(name, health);
+  }
+
+  bite(victim) {
+    if (victim instanceof Herbivore) {
+      if (!victim.hidden) {
```

Comment:
```suggestion
    if (victim instanceof Herbivore && !victim.hidden) {
```
You can just combine the conditions